src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -137,6 +131,31 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Configure the entire service set for this API. Descendants can override this method to choose
+        /// from the default service sets provided by RESTier.
+        /// </summary>
+        /// <param name="apiType">
+        /// CLR type of this <see cref="ApiBase"/>.
+        /// </param>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection"/> with which to create an <see cref="ApiConfiguration"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/>.</returns>
+        [CLSCompliant(false)]
+        protected virtual IServiceCollection ConfigureApiServices(Type apiType, IServiceCollection services)
+        {
+            var customizer = ApiConfiguration.Customize(apiType);
+            services.Apply(customizer.InnerMost);
+
+            return this.ConfigureApi(services)
+                .Apply(customizer.PrivateApi)
+                .UseAttributes(apiType)
+                .UseConventions(apiType)
+                .Apply(customizer.Overrides)
+                .Apply(customizer.OuterMost);
+        }
+
```

Comment:
I create a pull request per discussion locally, can you help to review and let us know your comments? thanks in advance for your great help.